OrbitCore/StringManager.cpp

```diff
@@ -0,0 +1,25 @@
+#include "StringManager.h"
+
+std::unique_ptr<StringManager> GStringManager;
+
+StringManager::StringManager() {}
+StringManager::~StringManager() {}
+
+void StringManager::Add(uint64_t key, const std::string& str) {
+  if (key_to_string_.count(key) == 0) {
+    key_to_string_.emplace(key, str);
+  }
+}
+
+std::string StringManager::Get(uint64_t key) {
```

Comment:
does std::optional make sense here? You can use value_or() in order to get empty string by default.